### PR TITLE
feat(perf-issues): Consider parent span in MN+1 DB detector

### PR DESCRIPTION
**Pull request by mjq-sentry**
Following the lead of the N+1 DB Queries detector, the MN+1 DB Queries detector should also consider the involved spans' parent.
* All spans must have the same parent
* That parent should be returned as part of the problem metadata
* The parent's op and hash should be included in the problem fingerprint

This will eliminate a few legit MN+1s (because of the same parent requirement), but the stricter fingerprinting will also fix inappropriate grouping in some scenarios.

### CODE REVIEW

File: src/sentry/utils/performance_issues/performance_detection.py [hunk 1]
```diff
@@ -1237,7 +1261,7 @@ class MNPlusOneDBSpanDetector(PerformanceDetector):
 
     def init(self):
         self.stored_problems = {}
-        self.state = SearchingForMNPlusOne(self.settings)
+        self.state = SearchingForMNPlusOne(self.settings, self.event())
```
Comment (gggritso) [inline src/sentry/utils/performance_issues/performance_detection.py:1264]: 👀 `self.event()`? Is `event` also a callable?

**Review by gggritso — COMMENTED** [review]
Makes sense! There's one bit that looks strange, but otherwise LGTM 👌🏻

File: src/sentry/utils/performance_issues/performance_detection.py (hunk 1 above)
Comment (mjq-sentry) [reply src/sentry/utils/performance_issues/performance_detection.py:1264]: It is indeed! https://github.com/getsentry/sentry/blob/master/src/sentry/utils/performance_issues/base.py#L102-L103 Render blocking assets uses it too, e.g. https://github.com/getsentry/sentry/blob/master/src/sentry/utils/performance_issues/performance_detection.py#L441

Comment (gggritso) [reply src/sentry/utils/performance_issues/performance_detection.py:1264]: Oohh, I see, okay!

**Review by gggritso — APPROVED** [review]

**Pull request merged**